File: Data Structures & Algorithms/non-cyclical-number/submission-3.py
class Solution:
    def isHappy(self, n: int) -> bool:
        numbersSeen = [n]
        while (True):
            if n == 1:
                return True
            temp = 0
            digits = len(str(n))
            for _ in range(digits):
                digit = n % 10
                temp += digit ** 2
                n = n // 10
            if temp in numbersSeen:
                return False
            numbersSeen.append(temp)
            n = temp
            

File: Data Structures & Algorithms/climbing-stairs/submission-3.cpp
class Solution {
public:
    int climbStairs(int n) {
        int temp1 = 1, temp2 = 1;
        for (int i = 1; i < n; i++) {
            int temp = temp2;
            temp2 += temp1;
            temp1 = temp;
        }
        return temp2;
    }
};
